File: game/src/main/java/com/scottlogic/hackathon/game/Move.java
package com.scottlogic.hackathon.game;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

/** A player's movement. */
@JsonDeserialize(as = MoveImpl.class)
public interface Move {
  /**
   * @return The id of the player carrying out the movement.
   */
  Id getPlayer();

  /**
   * @return The direction of the movement.
   */
  Direction getDirection();
}

File: python-contestant/src/game/move.py
from dataclasses import dataclass

from game.direction import Direction


@dataclass(frozen=True)
class Move:
    """
    Represents a player's movement.

    Attributes:
        player (int): The ID of the player carrying out the movement.
        direction (Direction): The direction of the movement.
    """
    player: int
    direction: Direction
